engine/common/synchronization/request_handler.go

```diff
@@ -375,37 +369,24 @@ func (r *RequestHandler) processAvailableRequests() error {
 	}
 }
 
-// requestProcessingLoop is a separate goroutine that performs processing of queued requests
-func (r *RequestHandler) requestProcessingLoop() {
+// requestProcessingWorker is a separate goroutine that performs processing of queued requests.
+// Multiple instances may be invoked. It is invoked and managed by the Engine or RequestHandlerEngine
+// which embeds this RequestHandler.
+func (r *RequestHandler) requestProcessingWorker(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
 	notifier := r.requestMessageHandler.GetNotifier()
+	done := ctx.Done()
```

Comment:
curious why you saved this as a variable?

Reply:
`Done` reads an `atomic.Value`, which is two atomic pointers under the hood. So there's hypothetically some benefit to not actually calling `Done` every time in a loop.

Out of curiosity, I benchmarked it and there is about a 20% difference (with one thread). But both are obviously very fast, way below the threshold for making a noticeable difference. Maybe in very tight loops, or when there are many more threads sharing the same context, it would be impactful.

Reply:
I think it's a nice optimization which doesn't introduce any complexity.